### PR TITLE
Rename `configparser.Parser` as `config.Map`

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

**Review by dashpole — APPROVED** [review]
configmap.ConfigMap stutters, but I don't have a better alternative to offer.

**Review by codeboten — APPROVED** [review]
> configmap.ConfigMap stutters, but I don't have a better alternative to offer.

I agree with this, the only way I could imagine this not sounding repetitive is by moving out of the `config` package altogether. Moving it to `config` makes it slightly better to use `config.ConfigMap` but I don't know if it's worth the effort.

**Comment by bogdandrutu** [comment]
@codeboten @dashpole tried to make this `config.Map`, this avoids all the duplications.

**Review by dashpole — APPROVED** [review]
Nice.  This is actually much easier to read, and not just because I don't confuse it with k8s configmaps anymore.

File: config/config.go [hunk 1]
```diff
@@ -203,10 +201,9 @@ type validatable interface {
 // Unmarshallable defines an optional interface for custom configuration unmarshaling.
 // A configuration struct can implement this interface to override the default unmarshaling.
 type Unmarshallable interface {
-	// Unmarshal is a function that un-marshals a Parser into the unmarshable struct in a custom way.
-	// componentSection *Parser
-	//   The config for this specific component. May be nil or empty if no config available.
-	Unmarshal(componentSection *configparser.ConfigMap) error
+	// Unmarshal is a function that un-marshals a config.Map into the unmarshable struct in a custom way.
```
Comment (codeboten) [inline config/config.go:204]: nit
```suggestion
	// Unmarshal is a function that unmarshals a config.Map into the unmarshable struct in a custom way.
```

File: config/configunmarshaler/defaultunmarshaler.go [hunk 2]
```diff
@@ -111,7 +110,7 @@ func NewDefault() ConfigUnmarshaler {
 
 // Unmarshal the Config from a Parser.
```
Comment (codeboten) [inline config/configunmarshaler/defaultunmarshaler.go:111]:
```suggestion
// Unmarshal the Config from a Map.
```

File: config/internal/configsource/manager.go [hunk 3]
```diff
@@ -185,8 +185,8 @@ func NewManager(_ *configparser.ConfigMap) (*Manager, error) {
 // Resolve inspects the given config.Parser and resolves all config sources referenced
 // in the configuration, returning a config.Parser fully resolved. This must be called only
 // once per lifetime of a Manager object.
```
Comment (codeboten) [inline config/internal/configsource/manager.go:187]:
```suggestion
// Resolve inspects the given config.Map and resolves all config sources referenced
// in the configuration, returning a config.Map fully resolved. This must be called only
// once per lifetime of a Manager object.
```

**Review by codeboten — APPROVED** [review]
I agree w/ @dashpole, this is great! Just a couple of nits in the comments

**Pull request merged**